Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -0,0 +1,1098 @@
+        borrowing_x.size.resize(box);
+        borrowing_x.size.setVal<amrex::RunOn::Host>(0);
+        amrex::Long ncells = box.numPts();
+        borrowing_x.inds.resize(8*ncells);
+        borrowing_x.neigh_faces.resize(8*ncells);
+        borrowing_x.area.resize(8*ncells);
+    }
+
+    idim = 1;
+    for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+        amrex::Box const &box = mfi.validbox();
+        auto &borrowing_y = (*m_borrowing[maxLevel()][idim])[mfi];
+        borrowing_y.inds_pointer.resize(box);
+        borrowing_y.size.resize(box);
+        borrowing_y.size.setVal<amrex::RunOn::Host>(0);
+        amrex::Long ncells = box.numPts();
+        borrowing_y.inds.resize(8*ncells);
+        borrowing_y.neigh_faces.resize(8*ncells);
+        borrowing_y.area.resize(8*ncells);
+    }
+
+    idim = 2;
+    for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+        amrex::Box const &box = mfi.validbox();
+        auto &borrowing_z = (*m_borrowing[maxLevel()][idim])[mfi];
+        borrowing_z.inds_pointer.resize(box);
+        borrowing_z.size.resize(box);
+        borrowing_z.size.setVal<amrex::RunOn::Host>(0);
+        amrex::Long ncells = box.numPts();
+        borrowing_z.inds.resize(8*ncells);
+        borrowing_z.neigh_faces.resize(8*ncells);
+        borrowing_z.area.resize(8*ncells);
+    }
+}
+
+/**
+* \brief For the face of cell pointing in direction idim, compute the number of faces
+* we need to intrude. For the one-way extension this function returns only one or zero: one if the
+* face can be extended withe the one-way extension, zeros if it can't.
+*/
+AMREX_GPU_DEVICE
+int
+WarpX::ComputeNBorrowOneFaceExtension(const amrex::Dim3 cell, const amrex::Real S_ext,
+                                      const amrex::Array4<amrex::Real>& S_red,
+                                      const amrex::Array4<int>& flag_info_face,
+                                      const amrex::Array4<int>& flag_ext_face, const int idim) {
+    const int i = cell.x;
+    const int j = cell.y;
+    const int k = cell.z;
+    int n_borrow = 0;
+    bool stop = false;
+    if(idim == 0){
+        for (int j_n = -1; j_n < 2; j_n++) {
+            for (int k_n = -1; k_n < 2; k_n++) {
+                //This if makes sure that we don't visit the "diagonal neighbours"
+                if (!(j_n == k_n or j_n == -k_n)) {
+                    // Here a face is available if it doesn't need to be extended itself and if its
+                    // area exceeds Sz_ext. Here we need to take into account if the intruded face
+                    // has given away already some area, so we use Sz_red rather than Sz.
+                    // If no face is available we don't do anything and we will need to use the
+                    // multi-face extensions.
+                    if (S_red(i, j + j_n, k + k_n) > S_ext
+                        and (flag_info_face(i, j + j_n, k + k_n) == 1 or
```

Comment:
Let's avoid `and` for `&&` and `not` for `!` to be compatible with more compilers (i.e., MSVC in certain conditions).
Similarly `or` -> `||`.

If/when we enable EB for the Windows build we should see this as CI error (#2170).

Reply:
Oh wow! I wasn't aware of this. I will change them all.